src/eth/decoder/types.ts

```diff
@@ -0,0 +1,34 @@
+import { addHexPrefix, bufferToHex } from 'ethereumjs-util';
+import { formatBool, isArray, parseArraySubType, parsePrimitive, Primitive } from '../../base/decoder/types';
+import { BigNumber } from 'bignumber.js';
+
+function formatArray(values: any[], subtype: string): any[] {
+  return values.map((value) => formatValue(value, subtype));
+}
+
+/**
+ * Format values of different types
+ * @param value The value to format
+ * @param type The solidity type string of the value
+ * @return The formatted value
+ */
+export function formatValue(value: any, type: string): any {
+  if (isArray(type)) {
+    return formatArray(value, parseArraySubType(type));
+  } else {
+    switch (parsePrimitive(type)) {
+      case Primitive.Address:
+        return addHexPrefix(value);
+      case Primitive.Bool:
+        return formatBool(value);
+      case Primitive.Bytes:
+        return bufferToHex(value);
+      case Primitive.Int:
+        // value is a Buffer
```

Comment:
let's type check this then and assert it

Reply:
There are type check for this here : https://github.com/BitGo/eth-contracts/blob/STLX-5_generic/src/base/decoder/types.ts#L40

Reply:
Is already type checked at the parser

Reply:
ok good